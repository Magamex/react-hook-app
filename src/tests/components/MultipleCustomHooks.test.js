import React from 'react';
import { shallow } from 'enzyme';
import { MultipleCustomHooks } from './../../components/examples/MultipleCustomHooks';
import { useFetch } from '../../Hooks/useFetch';
import { useCounter } from './../../Hooks/useCounter';

jest.mock('../../Hooks/useFetch');
jest.mock('./../../Hooks/useCounter');

describe('Pruebas en <MultipleCustomHooks/>', () => {
    

    test('debe de mostrarse correctamente', () => {
        useCounter.mockReturnValue({
            counter: 1,
            increment: () => {}
        })

        useFetch.mockReturnValue({
            data:null,
            loading: true,
            error: null
        });

        const wrapper = shallow(<MultipleCustomHooks/>);
        expect(wrapper).toMatchSnapshot();
    });

    test('Debe de mostrar la informacion', () => {
        useCounter.mockReturnValue({
            counter: 1,
            increment: () => {}
        })
        
        useFetch.mockReturnValue({
            data:[{
                author: 'Pepe',
                quote: 'Hola Mundo'
            }],
            loading: false,
            error: null
        });
        const wrapper = shallow(<MultipleCustomHooks/>);

        expect(wrapper.find('.loading').exists()).toBe(false);
        expect(wrapper.find('p').text().trim()).toBe('Hola Mundo');
        expect(wrapper.find('footer').text().trim()).toBe('Pepe');

        console.log(wrapper.html())
    });
});
