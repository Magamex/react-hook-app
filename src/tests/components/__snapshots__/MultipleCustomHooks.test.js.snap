// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <MultipleCustomHooks/> debe de mostrarse correctamente 1`] = `
<div>
  <h1>
    Multiple Custom Hooks
  </h1>
  <h2>
    breakingbad Quotes
  </h2>
  <hr />
  <div
    className="loading"
  >
    Loading...
  </div>
</div>
`;
